src/JsRoot/JsRootGraph2D.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react';
+import Skeleton from '@mui/material/Skeleton';
+import { useJSROOT } from './JsRootService';
+import { useVisible } from 'react-hooks-visible';
+import { Page2D } from './GraphData';
+
+export function JsRootGraph2D(props: Page2D) {
+	const { JSROOT } = useJSROOT();
+	const [containerEl, visible] = useVisible<HTMLDivElement>();
+
+	const [obj, setObj] = useState(undefined);
+	const [drawn, setDrawn] = useState(false);
+	const [isVisible, setIsVisible] = useState(false);
+
+	useEffect(() => {
+		setIsVisible(visible > 0.3);
+		return () => setIsVisible(false);
+	}, [visible]);
+
+	useEffect(() => {
+		if (drawn || !visible) return;
+		const x = props.first_axis.values;
+		const y = props.second_axis.values;
+		const z = props.data.values;
+
+		const nxpoints = x.length;
+		const nypoints = y.length;
+
+		const gr = JSROOT.createHistogram('TH2F', nxpoints, nypoints);
+
+		gr.fXaxis.fXmin = x[0];
+		gr.fXaxis.fXmax = x[nxpoints - 1];
+		gr.fXaxis.fTitle = `${props.first_axis.name} [${props.first_axis.unit}]`;
+
+		gr.fYaxis.fXmin = y[0];
+		gr.fYaxis.fXmax = y[nypoints - 1];
+		gr.fYaxis.fTitle = `${props.second_axis.name} [${props.second_axis.unit}]`;
+
+		gr.fTitle = `${props.data.name} [${props.data.unit}]`;
+
+		for (let x = 0; x < nxpoints; x++)
+			for (let y = 0; y < nypoints; y++) {
+				gr.setBinContent(gr.getBin(x + 1, y + 1), z[x + y * nxpoints]);
+			}
+
+		setObj(gr);
+		setDrawn(false);
+	}, [JSROOT, props, drawn, visible]);
+
+	useEffect(() => {
+		if (obj && !drawn && isVisible) {
+			JSROOT.cleanup(containerEl.current);
+			JSROOT.redraw(containerEl.current, obj, 'col');
```

Comment:
Can you try `COLZ` ?
According to
https://root.cern.ch/doc/v608/classTHistPainter.html

```
Same as "COL". In addition the color palette is also drawn. 
```